src/solver/conv_asm_3x3u.cpp

```diff
@@ -209,8 +213,12 @@ bool ConvAsm3x3U::IsApplicable(const ConvolutionContext& params) const
         && (params.n_inputs / params.group_counts) % 4 == 0 /// \todo: remove restriction that (n_inputs/group_counts) must be multiple of 4
         && params.in_width > 3
         && params.in_width <= 1000
-        && IN_BUF_SZ <= 4 * GIB
-        && OUT_BUF_SZ <= 4 * GIB
+        && ELEM_SZ * h_w < GIB
+        && ELEM_SZ * n_c < 4 * GIB
+        && ELEM_SZ * n_k < 4 * GIB
+        && ELEM_SZ * n_c_h_w < 256 * TIB
+        && ELEM_SZ * n_k_h_w < 256 * TIB
```

Comment:
Original `OUT_BUF_SZ` does not assume that `(out_H == in_H) && (out_W == in_W)` and looks clearer, right?

Reply:
You are right, thanks for the recommendation!